forwarder.go

```diff
@@ -10,24 +10,44 @@ import (
 	"net/http"
 	"os"
 	"strings"
+	"sync"
 )
 
+const workers = 8
+
 func main() {
 	log.Println("Splunk forwarder: Started")
 	defer log.Println("Splunk forwarder: Stopped")
 
+	forSplunk := make(chan string)
+
+	var wg sync.WaitGroup
+
+	for i := 0; i < workers; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for msg := range forSplunk {
+				postToSplunk(msg)					
+			}
+		}()
+	}
+
 	br := bufio.NewReader(os.Stdin)
 	for {
 		str, err := br.ReadString('\n')
 
 		if err != nil {
 			if err == io.EOF {
+				close(forSplunk)
 				return
 			}
 			log.Fatal(err)
 		}
-		postToSplunk(str)
+		forSplunk <- str
 	}
+
+	wg.Wait()
```

Comment:
With my limited knowledge of Go I cannot say whether this is true or false but according to this example https://golang.org/pkg/sync/#example_WaitGroup

    var wg sync.WaitGroup
    var urls = []string{
            "http://www.golang.org/",
            "http://www.google.com/",
            "http://www.somestupidname.com/",
    }
    for _, url := range urls {
            // Increment the WaitGroup counter.
            wg.Add(1)
            // Launch a goroutine to fetch the URL.
            go func(url string) {
                    // Decrement the counter when the goroutine completes.
                    defer wg.Done()
                    // Fetch the URL.
                    http.Get(url)
            }(url)
    }
    // Wait for all HTTP fetches to complete.
    wg.Wait()

So wg.Wait() sits outside of the for block. 

What's your view on where the wg.Wait() should be placed?